app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.setupquestions.annualallowance
+
+import controllers.actions._
+import forms.setupquestions.annualallowance.NetIncomeAbove190KIn2023FormProvider
+import models.Mode
+import models.tasklist.sections.SetupSection
+import pages.setupquestions.annualallowance.NetIncomeAbove190KIn2023Page
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.UserDataService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.setupquestions.annualallowance.NetIncomeAbove190KIn2023View
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class NetIncomeAbove190KIn2023Controller @Inject() (
+  override val messagesApi: MessagesApi,
+  userDataService: UserDataService,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: NetIncomeAbove190KIn2023FormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: NetIncomeAbove190KIn2023View
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    val preparedForm = request.userAnswers.get(NetIncomeAbove190KIn2023Page) match {
+      case None        => form
+      case Some(value) => form.fill(value)
+    }
+
+    Ok(view(preparedForm, mode))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          value =>
+            for {
+              updatedAnswers <- Future.fromTry(request.userAnswers.set(NetIncomeAbove190KIn2023Page, value))
+              redirectUrl     = NetIncomeAbove190KIn2023Page.navigate(mode, updatedAnswers).url
+              answersWithNav  = SetupSection.saveNavigation(updatedAnswers, redirectUrl)
+              _              <- userDataService.set(answersWithNav)
+            } yield Redirect(redirectUrl)
+        )
+  }
```

Comment:
Needs to be able to update kickout statuses to 2 when yes

Reply:
Completely forgot about statuses. Added it now